### DISCUSSION TITLE
jshell window

### DISCUSSION THREAD

**Discussion by sboesebeck** [mfussenegger/nvim-jdtls#403] [General]
Hi,

I switched to nvim with jdtls as main IDE for java a couple of months ago. So far it is working fine. 

I was using `jshell` in my project (a really great feature) and noticed that the buffer it is opened in is kind of a temporary one. It means, when switching back to another buffer, the `jshell` is just gone. If I open it, a new shell is spawned.

Is that something wrong with my config? Did I miss something or is that working as it is supposed to work? If so, would that be a complex feature to implement?

**Comment by mfussenegger** [comment]
It opens jshell in a terminal buffer, using the current window.
If you switch to another buffer it should still keep the `jshell` around, unless you're using `nohidden`.

You could use something like `:sp | JdtJshell` to ensure it gets its own window.

**Comment by sboesebeck** [comment]
when I look at my processes, the jshells are indeed still running. But I cannot get them "back"!

But your tip works great - thanks, that solved it!

**Reply by mfussenegger** [reply]
I just realized that `buflisted` wasn't set. So it didn't appear in `buffers` or when switching with `:bnext`. (I use https://github.com/mfussenegger/nvim-qwahl to switch buffers, and it by defaults includes unlisted buffers)

Can you try with https://github.com/mfussenegger/nvim-jdtls/pull/404 ?

**Comment by sboesebeck** [comment]
I just tried it - now it is listed as a standard buffer and I can switch back and forth... Thanks!
